plugins/module_utils/hmc_resource.py

```diff
@@ -482,21 +482,53 @@ def fetchIODetailsForNetboot(self, nimIP, gateway, lparIP, viosName, profName, s
         result = self.hmcconn.execute(lpar_netboot)
         return self._parseIODetailsFromNetboot(result)
 
-    def installOSFromNIM(self, loc_code, nimIP, gateway, lparIP, vlanID, vlanPrio, submask, viosName, profName, systemName):
-        lpar_netboot = self.CMD['LPAR_NETBOOT'] +\
-            self.OPT['LPAR_NETBOOT']['-F'] +\
+    def installOSFromNIM(self, loc_code, nimIP, gateway, lparIP, vlanID, vlanPrio, submask, viosName, profName, systemName, lparMac=None):
+        if loc_code:
+            os_command = self.OPT['LPAR_NETBOOT']['-L'] + loc_code +\
+                self.OPT['LPAR_NETBOOT']['-V'] + vlanID +\
+                self.OPT['LPAR_NETBOOT']['-Y'] + vlanPrio
+        elif lparMac:
+            os_command = self.OPT['LPAR_NETBOOT']['-x'] +\
+                self.OPT['LPAR_NETBOOT']['-v'] +\
+                self.OPT['LPAR_NETBOOT']['-i'] +\
+                self.OPT['LPAR_NETBOOT']['-s'] + "auto" +\
+                self.OPT['LPAR_NETBOOT']['-d'] + "auto" +\
+                self.OPT['LPAR_NETBOOT']['-m'] + lparMac
+        else:
+            pass
+
+        lpar_netboot = self.CMD['LPAR_NETBOOT'] + self.OPT['LPAR_NETBOOT']['-F'] +\
             self.OPT['LPAR_NETBOOT']['-D'] +\
             self.OPT['LPAR_NETBOOT']['-T'] + "ent" +\
-            self.OPT['LPAR_NETBOOT']['-L'] + loc_code +\
             self.OPT['LPAR_NETBOOT']['-S'] + nimIP +\
             self.OPT['LPAR_NETBOOT']['-G'] + gateway +\
             self.OPT['LPAR_NETBOOT']['-C'] + lparIP +\
-            self.OPT['LPAR_NETBOOT']['-V'] + vlanID +\
-            self.OPT['LPAR_NETBOOT']['-Y'] + vlanPrio +\
             self.OPT['LPAR_NETBOOT']['-K'] + submask +\
+            os_command +\
             " " + viosName + " " + profName + " " + systemName
         self.hmcconn.execute(lpar_netboot)
 
+    def getconsolelog(self, module, lpar_hmc, userid, hmc_password, systemName, viosName):
+        conn = HmcCliConnection(module, lpar_hmc, userid, hmc_password)
+        cmd = 'rmvterm -m ' + systemName + ' -p ' + viosName
+        conn.execute(cmd)
+        cmd = 'mkvterm -m ' + systemName + ' -p ' + viosName
+        stdout = conn.execute(cmd)
+        try:
+            for line in stdout:
+                logger.debug(line.strip())
+                logger.debug("\n")
+        except UnicodeDecodeError:
+            pass
+
+    def checkconsolelog(self, module, lpar_ip, lpar_hmc, userid, hmc_password, systemName, viosName):
```

Comment:
Are we passing vios_name here?, if no please change the parameter name to lpar_name

Reply:
changed to lparName